### PR TITLE
PYIC-4115: Added Experian KBV consumer side contract tests

### PR DESCRIPTION
**Pull request by RogVanGDS**
### What changed

Added Experian KBV consumer side contract tests

### Why did it change

To unblock Experian KBV producer side tests

### Issue tracking

- [PYIC-4115](https://govukverify.atlassian.net/browse/PYIC-4115)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

[PYIC-4115]: https://govukverify.atlassian.net/browse/PYIC-4115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java [hunk 1]
```diff
@@ -0,0 +1,1041 @@
...
+                       "postalCode": "HP16 0AL",
+                       "addressCountry": "GB"
+                     }
+                   ]
+             },
+             "evidence": [
+                 {
+                   "type": "IdentityCheck",
+                   "txn": "dummyTxn",
+                   "verificationScore": 0,
```
Comment (DanCorderIPV) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java:230]: I know that this example has come from Confluence, but I'm suspicious of this verification score as this should be a successful VC. This table suggest that the score should be 2 in this case: https://govukverify.atlassian.net/wiki/spaces/OJ/pages/3580100670/Experian+KBV+question+policy+-+WIP#Current-Implementation-(with-Contra-Indicators)

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java [hunk 2]
```diff
@@ -0,0 +1,1041 @@
...
+                        "failedCheckDetails": [
+                            {
+                                "kbvResponseMode": "free_text",
+                                "checkMethod": "kbv"
+                            },
+                            {
+                                "kbvResponseMode": "multiple_choice",
+                                "checkMethod": "kbv"
+                            }],
+                            "ci": ["V03"]
```
Comment (DanCorderIPV) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java]: Whilst it is OK to use CI codes with no context, we should default to using fake codes unless the provider can't produce VCs with fake codes.

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java [hunk 3]
```diff
@@ -0,0 +1,1041 @@
...
+                        }
+                      ]
+                    },
+                "evidence": [
+                    {
+                      "checkDetails": [
+                        {
+                          "checkMethod": "kbv",
+                          "kbvQuality": 2,
+                          "kbvResponseMode": "free_text"
```
Comment (blakeyp) [inline lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java]: Not sure where this has come from but the Experian KBV qs are all multiple choice

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java (hunk 1 above)
Comment (blakeyp) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java:230]: Yeah a KBV fail would be zero scoring with a V03 CI (the one below)
But there is another scenario - Experian KBV uses a zero scoring VC without a CI to denote abandon/thin file

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java (hunk 3 above)
Comment (DanCorderIPV) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java]: Looks like it came from here: https://github.com/govuk-one-login/architecture/blob/main/rfc/0024-identity-vc-for-checks.md#knowledge-based-verification-kbv-check which where the Orange team pointed Roger.
I've had a quick search in Confluence and can't see any more specific Experian example VCs

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java (hunk 1 above)
Comment (DanCorderIPV) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java:230]: That sounds like an additional scenario worth testing, this VC has too many questions to be thin file or abandoned though :)

File: lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java (hunk 3 above)
Comment (blakeyp) [reply lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/experianKbvCri/ContractTest.java]: Yeah the RFC is just describing a generic KBV CRI. I'd probably change it to multiple choice for now and can be something to come back to if we need to?

**Review by blakeyp — APPROVED** [review]

**Pull request merged**